hadoop-hdfs-project/hadoop-hdfs-client/src/main/java/org/apache/hadoop/hdfs/protocol/datatransfer/sasl/SaslDataTransferClient.java

```diff
@@ -603,7 +603,17 @@ private IOStreamPair doSaslHandshake(InetAddress addr,
           conf, cipherOption, underlyingOut, underlyingIn, false) :
           sasl.createStreamPair(out, in);
     } catch (IOException ioe) {
-      sendGenericSaslErrorMessage(out, ioe.getMessage());
+      try {
+        sendGenericSaslErrorMessage(out, ioe.getMessage());
+      } catch (Exception ioe2) {
```

Comment:
Given we catch `Exception`, I think `ioe2` should be `e`.

Reply:
Oh, thanks. fixed